build.ps1

```diff
@@ -105,10 +106,22 @@ else {
 Invoke-BuildStep 'Running Restore' {
 
     # Restore
-    Trace-Log ". `"$MSBuildExe`" build\build.proj /t:EnsurePackageReferenceVersionsInSolution /p:Configuration=$Configuration"
-    & $MSBuildExe build\build.proj /t:EnsurePackageReferenceVersionsInSolution /p:Configuration=$Configuration
-    Trace-Log ". `"$MSBuildExe`" build\build.proj /t:RestoreVS /p:Configuration=$Configuration /p:ReleaseLabel=$ReleaseLabel /p:BuildNumber=$BuildNumber /v:m /m:1"
-    & $MSBuildExe build\build.proj /t:RestoreVS /p:Configuration=$Configuration /p:ReleaseLabel=$ReleaseLabel /p:BuildNumber=$BuildNumber /v:m /m:1
+    $args = "build\build.proj", "/t:EnsurePackageReferenceVersionsInSolution", "/p:Configuration=$Configuration"
+    if ($Binlog)
+    {
+        $args += "-bl:msbuild.ensurepr.binlog"
+    }
+
+    Trace-Log ". `"$MSBuildExe`" $args"
+    & $MSBuildExe @args
+
+    $args = "build\build.proj", "/t:RestoreVS", "/p:Configuration=$Configuration", "/p:ReleaseLabel=$ReleaseLabel", "/p:BuildNumber=$BuildNumber", "/v:m", "/m:1"
+    if ($Binlog)
+    {
+        $args += "-bl:msbuild.restore.binlog"
```

Comment:
Trying to understand how this works... I thought it was just concatenating a filename suffix and extension for the binlog. But this one in particular seems it would be concatenating to `/m:1` ?

Reply:
`$args` is an array, so `+=` is appending this string to the array.

Reply:
for what it's worth, I copied the same pattern as what we were already doing for strong name certificates: https://github.com/NuGet/NuGet.Client/blob/4dbf16e68eabf507b9f156c801b1b7c3b0586e43/build.ps1#L124-L130